py_experimenter/experimenter.py

```diff
@@ -256,94 +265,140 @@ def fill_table_with_rows(self, rows: List[dict]) -> None:
+        if `random_order` = True). Then the `experiment_function` is executed on its keyfield values defined in the table.
+        In this function the status of the experiment is continuously updated. The experiment can have the following states:
 
         * `running` when the execution of the experiment has been started, but not yet finished.
         * `error` if an exception was raised during the execution of the experiment.
         * `done` if the execution of the experiment has finished successfully.
 
+        If an error occurs during the execution of the experiment, it is logged to the `error` column in the table. If an 
+        error occurs before or after the execution of the `experiment_function`, it is logged via the function `_handle_error`.
+
         :param experiment_function: The function that should be executed with the different parametrizations.
         :type experiment_function: Callable[[dict, dict, ResultProcessor], None]
-        :param custom_fields: The custom field values to execute the `experiment_function` with.
-        :type custom_fields: dict
-        :param keyfields: The keyfield values to execute the `experiment_function` with. 
-        :type keyfields: dict
-        :param result_processor: The `ResultProcessor` that is responsible to update the database table.
-        :type result_processor: ResultProcessor
+        :param random_order: Indicates whether the experiments to be executed are chosen consecutively by its experiment
+                                ID (`False`) or in a randomized fashion (`True`). Defaults to False.
+        :type random_order: bool
+        :raises NoExperimentsLeftError: If there are no experiments left to be executed.
+        :raises DatabaseConnectionError: If an error occurred during the connection to the database.
         """
-        if result_processor._not_executed_yet():
-            result_processor._change_status('running')
-            result_processor._set_name(self.name)
-            result_processor._set_machine(socket.gethostname())
-            try:
-                logging.debug(f"Start of experiment_function on process {socket.gethostname()}")
-                experiment_function(keyfields, result_processor, custom_fields)
-            except Exception:
-                error_msg = traceback.format_exc()
-                logging.error(error_msg)
-                result_processor._write_error(error_msg)
-                result_processor._change_status('error')
-            else:
-                result_processor._change_status('done')
+        result_field_names = utils.get_result_field_names(self.config)
+        if self.has_section('CUSTOM'):
+            custom_fields = dict(self.config.items('CUSTOM'))
+        else:
+            custom_fields = None
+        table_name = self.get_config_value('PY_EXPERIMENTER', 'table')
+
+        experiment_id, keyfield_values = self.dbconnector.get_experiment_configuration(random_order)
```

Comment:
why is the `experiment_id` not further used?

Reply:
The experiment_id is added so that the rework of the result_processor requires less effort.